components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/OperationPolicySpecification.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.wso2.carbon.apimgt.api.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class OperationPolicySpecification {
+
+    public enum PolicyCategory {
+        Mediation,
+        Security
+    }
+
+    private PolicyCategory category = PolicyCategory.Mediation;
+    private String name;
+    private String version = "v1";
+    private String displayName;
+    private String description;
+    private List<String> applicableFlows = new ArrayList<>();
+    private List<String> supportedGateways = new ArrayList<>();
+    private List<String> supportedApiTypes = new ArrayList<>();
+    private List<OperationPolicySpecAttribute> policyAttributes = new ArrayList<>();
+    private boolean multipleAllowed;
+
+    public String getName() {
+
+        return name;
+    }
+
+    public void setName(String name) {
+
+        this.name = name;
+    }
+
+    public String getVersion() {
+
+        return version;
+    }
+
+    public void setVersion(String version) {
+
+        this.version = version;
+    }
+
+    public String getDisplayName() {
+
+        return displayName;
+    }
+
+    public void setDisplayName(String displayName) {
+
+        this.displayName = displayName;
+    }
+
+    public String getDescription() {
+
+        return description;
+    }
+
+    public void setDescription(String description) {
+
+        this.description = description;
+    }
+
+    public List<String> getApplicableFlows() {
+
+        return applicableFlows;
+    }
+
+    public void setApplicableFlows(List<String> applicableFlows) {
+
+        this.applicableFlows = applicableFlows;
+    }
+
+    public List<String> getSupportedGateways() {
+
+        return supportedGateways;
+    }
+
+    public void setSupportedGateways(List<String> supportedGateways) {
+
+        this.supportedGateways = supportedGateways;
+    }
+
+    public List<String> getSupportedApiTypes() {
+
+        return supportedApiTypes;
+    }
+
+    public void setSupportedApiTypes(List<String> supportedApiTypes) {
+
+        this.supportedApiTypes = supportedApiTypes;
+    }
+
+    public List<OperationPolicySpecAttribute> getPolicyAttributes() {
+
+        return policyAttributes;
```

Comment:
should we return an immutable list ?

Reply:
Looks like we normally return a list as added here, with similar models (refer [1]). If we were to return immutable lists, the standard maintained across such models may vary. Do you think we should switch to an immutable list nevertheless?

[1] https://github.com/wso2/carbon-apimgt/tree/master/components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model

Reply:
depends on the use case, if we expect it to be not modified, then safe to return an immutable one